### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by acron0**

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -1,15 +1,25 @@
 (defproject witan.workspace "0.1.0-SNAPSHOT"
   :description "FIXME: write description"
-  :url "http://example.com/FIXME"
-  :license {:name "Eclipse Public License"
-            :url "http://www.eclipse.org/legal/epl-v10.html"}
   :dependencies [[org.clojure/clojure "1.7.0"]
                  [org.clojure/core.async "0.2.371"]
                  [com.stuartsierra/component "0.3.1"]
+                 [com.taoensso/timbre "4.3.1"]
```
Comment (elisehuard) [inline project.clj]: would be good to look into how to get logstash format out of timbre if it is your logging framework of choice

File: src/witan/workspace/handler.clj [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  (api
+   {:swagger
+    {:ui "/"
+     :spec "/swagger.json"
+     :data {:info {:title "Witan Workspace Query service"
+                   :description ""}
+            :tags [{:name "api", :description "some apis"}]}}}
+
+   (GET "/by-owner" req
+        :summary "Just a test"
```
Comment (elisehuard) [inline src/witan/workspace/handler.clj:24]: maybe less generic value for :summary?

File: src/witan/workspace/system.clj [hunk 3]
```diff
@@ -3,20 +3,38 @@
...
+            [witan.workspace.components.cassandra :refer [new-cassandra-connection]]
+            [witan.workspace.components.server :refer [new-http-server]]
             [witan.workspace.command          :refer [command-receiver]]
-            [witan.workspace.event            :refer [event-receiver]]))
+            [witan.workspace.event            :refer [event-receiver]])
+  (:gen-class))
 
 (defn new-system
   []
   (let [config {:kafka {:zk {:host "127.0.0.1"
```
Comment (elisehuard) [inline src/witan/workspace/system.clj]: edn file (probably on your todo)

File: src/witan/workspace/util.clj [hunk 4]
```diff
@@ -32,3 +32,22 @@
        (drop 1)
        (butlast)
        (apply str)))
+
+(defn replacer
+  "Calls  replacement function on different types"
+  [rfn x]
+  (condp = (type x)
+    clojure.lang.Keyword (-> x name rfn keyword)
+    clojure.lang.MapEntry (update x 0 (partial replacer rfn))
```
Comment (elisehuard) [inline src/witan/workspace/util.clj:41]: clever, this.  which use case does the MapEntry solve? (curiosity)

File: src/witan/workspace/protocols.clj [hunk 5]
```diff
@@ -2,3 +2,18 @@
...
+
+(defprotocol Database
+  (drop-table!
+    [this table])
+  (create-table!
+    [this table columns])
+  (insert!
+    [this table row]
+    [this table row args])
+  (select*
```
Comment (elisehuard) [inline src/witan/workspace/protocols.clj:16]: Not super convinced that this presents a huge plus over hayt's dsl, and we only have the one Database atm.  What was your motivation?

File: src/witan/workspace/util.clj (hunk 4 above)
Comment (acron0) [inline src/witan/workspace/util.clj:41]: It's so you can call `(map hyphen->underscore m)` on a hashmap. The motivation is to not have to interchange between `:kebab-case` and `:snake_case` as we go in and out of Cassandra.

Comment (elisehuard) [inline src/witan/workspace/util.clj:41]: 👍

File: src/witan/workspace/protocols.clj (hunk 5 above)
Comment (acron0) [inline src/witan/workspace/protocols.clj:16]: It's more to appease the component (as in stuartsierra's) isolation and also gives us the ability to use those replacer calls, mentioned above.

File: src/witan/workspace/handler.clj (hunk 2 above)
Comment (acron0) [inline src/witan/workspace/handler.clj:24]: I will improve this once it actually does something. Promise.

**Comment by elisehuard** [comment]
:shipit:

**Pull request merged**